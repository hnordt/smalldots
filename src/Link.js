import React, { PropTypes } from 'react'
import BrowserHistoryService, { locationPropTypes } from './BrowserHistoryService'

const Link = ({ location, onClick, ...props }) => (
  <BrowserHistoryService>
    {({ push }) => (
      <a
        {...props}
        href="#"
        onClick={event => {
          event.preventDefault()
          if (location) {
            push(location)
          }
          if (onClick) {
            onClick(event)
          }
        }}
      />
    )}
  </BrowserHistoryService>
)

Link.propTypes = {
  location: locationPropTypes,
  onClick: PropTypes.func
}

export default Link
