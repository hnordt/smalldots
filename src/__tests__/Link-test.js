import React from 'react'
import renderer from 'react-test-renderer'
import Link from '../Link'

test('renders correctly', () => {
  const tree = renderer.create(
    <Link to="/foo">
      Foo
    </Link>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
