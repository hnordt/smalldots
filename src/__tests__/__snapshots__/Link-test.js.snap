exports[`test renders correctly 1`] = `
<a
  href="#"
  onClick={[Function]}
  to="/foo">
  Foo
</a>
`;
